packages/esm-patient-chart-app/src/ui-components/action-menu.component.tsx

```diff
@@ -21,35 +19,29 @@ export const CHARTS_ACTION_MENU_ITEMS_SLOT = 'action-menu-items-slot';
 export const ActionMenu: React.FC<ActionMenuInterface> = ({ open }) => {
   const { t } = useTranslation();
   const layout = useLayoutType();
-  const { screenMode, active } = useWorkspace();
-  const { openWindows, updateWindowSize, windowSize } = useContextWorkspace();
+  const { windowSize, openWindows, updateWindowSize, screenMode } = useContextWorkspace();
 
-  const checkViewMode = () => {
-    if (active) {
-      if (windowSize.size === ScreenModeTypes.maximize) {
-        updateWindowSize(ScreenModeTypes.hide);
-      } else if (windowSize.size === ScreenModeTypes.normal) {
-        updateWindowSize(ScreenModeTypes.hide);
-      } else {
-        updateWindowSize(screenMode);
+  const checkViewMode = useCallback(
```

Comment:
A function called `check...` I expect not to have any side-effects. Please give this function a name that describes what it does.